flipt-client-go/client.go

```diff
@@ -77,213 +69,200 @@ type EvaluationClient struct {
 	if err != nil {
-		cerr = fmt.Errorf("can't instantiate Wasm module: %w", err)
-		return
+		return nil, fmt.Errorf("can't instantiate Wasm module: %w", err)
 	}
 
 	ctx, cancel := context.WithCancel(ctx)
 
-	c := &EvaluationClient{
+	c := &Client{
 		runtime: runtime,
 		mod:     mod,
 
-		// default values
-		namespace:      defaultNamespace,
-		baseURL:        "http://localhost:8080",
-		updateInterval: 2 * time.Minute, // default 120 seconds
-		errorStrategy:  ErrorStrategyFail,
-		fetchMode:      FetchModePolling,
-		requestTimeout: 30 * time.Second, // default timeout
+		cfg: cfg,
 
 		cancel:       cancel,
 		errChan:      make(chan error, 1),
 		snapshotChan: make(chan snapshot, 1),
 	}
 
-	c.httpClient = &http.Client{
-		Transport: &http.Transport{
-			DialContext: (&net.Dialer{
-				Timeout:   10 * time.Second,
-				KeepAlive: 5 * time.Second, // More aggressive TCP keepalive
-			}).DialContext,
-			IdleConnTimeout:       60 * time.Second,
-			TLSHandshakeTimeout:   10 * time.Second,
-			ResponseHeaderTimeout: 10 * time.Second,
-			ExpectContinueTimeout: 1 * time.Second,
-			ForceAttemptHTTP2:     true,
-			MaxIdleConns:          100,
-			MaxIdleConnsPerHost:   10,
-			// Enable HTTP/2 ping frames and other settings
-			Proxy:           http.ProxyFromEnvironment,
-			ReadBufferSize:  32 * 1024,
-			WriteBufferSize: 32 * 1024,
-			// HTTP/2 specific settings
-			MaxResponseHeaderBytes: 4 * 1024, // 4KB
-			DisableCompression:     false,    // Enable compression
-		},
-	}
-
-	for _, opt := range opts {
-		opt(c)
+	if c.cfg.HTTPClient == nil {
+		c.cfg.HTTPClient = defaultHTTPClient
 	}
 
-	if c.namespace == "" {
-		cerr = fmt.Errorf("namespace cannot be empty")
-		return
+	if c.cfg.Namespace == "" {
+		return nil, fmt.Errorf("namespace cannot be empty")
 	}
 
-	if c.baseURL == "" {
-		cerr = fmt.Errorf("baseURL cannot be empty")
-		return
+	if c.cfg.URL == "" {
+		return nil, fmt.Errorf("baseURL cannot be empty")
 	}
 
-	if c.fetchMode == FetchModePolling {
-		if c.updateInterval < 1*time.Second {
-			cerr = fmt.Errorf("update interval must be greater than 1 second")
-			return
+	if c.cfg.FetchMode == FetchModePolling {
+		if c.cfg.UpdateInterval < 1*time.Second {
+			return nil, fmt.Errorf("update interval must be greater than 1 second")
 		}
 
-		if c.requestTimeout < 1*time.Second {
-			cerr = fmt.Errorf("request timeout must be greater than 1 second")
-			return
+		if c.cfg.RequestTimeout < 1*time.Second {
+			return nil, fmt.Errorf("request timeout must be greater than 1 second")
 		}
 	}
 
 	// Store the base URL without trailing slash
-	c.baseURL = strings.TrimRight(c.baseURL, "/")
+	c.cfg.URL = strings.TrimRight(c.cfg.URL, "/")
 
 	// Set timeout only for polling mode
-	if c.fetchMode == FetchModePolling {
-		c.httpClient.Timeout = c.requestTimeout
+	if c.cfg.FetchMode == FetchModePolling {
+		c.cfg.HTTPClient.Timeout = c.cfg.RequestTimeout
```

Comment:
this could be unexpected thing for those people who use `WithHttpClient`. They probably may want to control it in and out... Maybe it will be good to include in godoc for `WithHttpClient`